instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/NetClientAttributesGetter.java

```diff
@@ -27,5 +27,22 @@
   Integer peerPort(REQUEST request, @Nullable RESPONSE response);
 
   @Nullable
-  String peerIp(REQUEST request, @Nullable RESPONSE response);
+  default String sockPeerAddr(REQUEST request, @Nullable RESPONSE response) {
+    return null;
+  }
+
+  @Nullable
+  default Integer sockPeerPort(REQUEST request, @Nullable RESPONSE response) {
+    return null;
+  }
+
+  @Nullable
+  default String sockFamily(REQUEST request, @Nullable RESPONSE response) {
+    return null;
+  }
+
+  @Nullable
+  default String sockPeerName(REQUEST request, @Nullable RESPONSE response) {
+    return null;
+  }
```

Comment:
thoughts on defaults or not for the `sock*` methods?

my thought was that these will be much less implemented, but we have not been adding default methods previously, so probably should not be defaults to force people to think about them(?)

Reply:
my 2c: this is an advanced scenario, not always feasible, and also with relatively small demand. If it's a bit less discoverable it's ok - it's easy to override the method later when instrumentation authors know why they need it.